src/api/helpers.ts

```diff
@@ -13,6 +17,84 @@ export async function getAuthenticationHeaders(options?: RequestInit): Promise<H
   return {}
 }
 
+interface JWTToken {
+  exp: number
+  fresh: false
+  iat: number
+  identity: string
+  jti: string
+  nbf: number
+  type: string
+}
+
+/**
+ * Implements a retry-pattern for http calls.
+ * Ignores native/v1/refresh_access_token.
+ *
+ * First decodes the local access token:
+ * on success: continue to the call
+ * on error (401): try to refresh the access token
+ * on error (other): propagates error
+ */
+export const retryableFetch = async (url: string, options: RequestInit): Promise<Response> => {
+  // dont ask a new token for this specific api call
+  if (url.includes('native/v1/refresh_access_token')) {
+    return await fetch(url, options)
+  }
+
+  // @ts-ignore
+  const authorizationHeader = options.headers?.['Authorization'] || ''
+  const token = authorizationHeader?.replace('Bearer ', '')
+
+  try {
+    const tokenContent = jwtDecode(token) as JWTToken
+
+    if (tokenContent.exp * 1000 <= new Date().getTime()) {
+      throw new Error('Token expired')
+    }
+  } catch (error) {
+    const newAcessToken = await refreshToken()
+
+    options = {
+      ...options,
+      headers: {
+        ...options.headers,
+        Authorization: `Bearer ${newAcessToken}`,
+      },
+    }
+  }
+
+  return await fetch(url, options)
+}
+
+/**
+ * Calls Api to refresh the access token using the in-keychain stored refresh token
+ * - on success: Stores the new access token
+ * - on error (422): propagates error
+ * - on error (other): throws an exception
+ */
+export const refreshToken = async (): Promise<string | null> => {
+  const keyChainRefreshToken = await getRefreshToken()
```

Comment:
```suggestion
  const refreshToken = await getRefreshToken()
```
non ? il n'y en a qu'un

Reply:
refreshToken portait un peu à confusion avec le nom de la fonction.

finalement je change le nom de la fonction: refreshAccessToken